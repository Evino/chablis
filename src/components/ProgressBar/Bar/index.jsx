import styled from 'styled-components';
import { colors, sizes } from '../../../styles/variables';

const Bar = styled.span`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 6px;
  margin-right: 0.6%;
  opacity: 0.2;
  position: relative;

  @media(min-width: 769px) {
    margin-right: 0.2%;
  }

  :nth-child(1) {
    background-color: ${colors.ORANGE_1};
    border-top-left-radius: ${sizes.RADIUS};
    border-bottom-left-radius: ${sizes.RADIUS};
  }

  :nth-child(2) {
    background-color: ${colors.ORANGE_2};
  }

  :nth-child(3) {
    background-color:  ${colors.ORANGE_3};
  }

  :nth-child(4) {
    background-color:  ${colors.ORANGE_4};
  }

  :nth-child(5) {
    background-color:  ${colors.ORANGE_5};
    border-top-right-radius: ${sizes.RADIUS};
    border-bottom-right-radius: ${sizes.RADIUS};
  }

  :nth-child(-n+${({ active }) => active}) {
    opacity: 1;
  }
`;

export default Bar;
