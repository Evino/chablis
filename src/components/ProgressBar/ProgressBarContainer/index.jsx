import styled from 'styled-components';

const ProgressBarContainer = styled.div`
  display: flex;
  align-items: center;

  p {
    font-size: 14px;
    margin-right: 10px;
    width: ${({ labelWidth }) => labelWidth || '60px'}
  }
`;

export default ProgressBarContainer;
