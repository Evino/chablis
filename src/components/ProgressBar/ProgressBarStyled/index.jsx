import styled, { css } from 'styled-components';
import { colors, zIndexes } from '../../../styles/variables';

const ProgressBarStyled = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;

  span:nth-child(${({ active }) => active}) {
    &:after {
      content: "";
      display: block;
      width: 10px;
      height: 10px;
      background-color: ${colors.WHITE};
      border-radius: 50%;
      border: 3px solid ${colors.ORANGE_4};
      position: absolute;
      right: -10px;
      z-index: ${zIndexes.STRATOSPHERE};
    }
  }

  ${({ active }) => active === 0 && css`
    span:first-child {
      &:after {
        content: '';
        display: block;
        width: 10px;
        height: 10px;
        background-color: ${colors.WHITE};
        border-radius: 50%;
        border: 3px solid ${colors.ORANGE_4};
        position: absolute;
        left: 0;
        z-index: ${zIndexes.STRATOSPHERE};
      }
    }
  `}`;

export default ProgressBarStyled;
