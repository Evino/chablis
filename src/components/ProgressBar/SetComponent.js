import React from 'react';
import PropTypes from 'prop-types';
import { Bar, ProgressBarStyled, ProgressBarContainer } from '.';
import BodySmall from '../Typography/BodySmall';

const ProgressBar = ({ active, label }) => (
  <ProgressBarContainer>
    {label && <BodySmall>{label}</BodySmall>}
    <ProgressBarStyled active={active}>
      {Array.from(Array(5), (_, i) => <Bar key={i} active={active} />)}
    </ProgressBarStyled>
  </ProgressBarContainer>
);

ProgressBar.propTypes = {
  active: PropTypes.number,
  label: PropTypes.string,
};

export default ProgressBar;
