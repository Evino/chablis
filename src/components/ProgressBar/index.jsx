import styled, { css } from 'styled-components';
import { sizes, zIndexes, colors } from '../../styles/variables';

export const Bar = styled.span`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 6px;
  margin-right: 0.2%;
  opacity: 0.2;
  position: relative;

  &:nth-child(1) {
    background-color: ${colors.ORANGE_1};
    border-top-left-radius: ${sizes.RADIUS};
    border-bottom-left-radius: ${sizes.RADIUS};
  }

  &:nth-child(2) {
    background-color: ${colors.ORANGE_2};
  }

  &:nth-child(3) {
    background-color:  ${colors.ORANGE_3};
  }

  &:nth-child(4) {
    background-color:  ${colors.ORANGE_4};
  }

  &:nth-child(5) {
    background-color:  ${colors.ORANGE_5};
    border-top-right-radius: ${sizes.RADIUS};
    border-bottom-right-radius: ${sizes.RADIUS};
  }

  &:nth-child(-n+${({ active }) => active}) {
    opacity: 1;
  }
`;

export const ProgressBarStyled = styled.div`
  display: flex;
  justify-content: space-between;
  width: 100%;

  & span:nth-child(${({ active }) => active}) {
    &:after {
      content: "";
      display: block;
      width: 10px;
      height: 10px;
      background-color: #fff;
      border-radius: 50%;
      border: 3px solid #ec720b;
      position: absolute;
      right: -10px;
      z-index: ${zIndexes.STRATOSPHERE};
    }
  }

  ${({ active }) => active === 0 && css`
    & span:first-child {
      &:after {
        content: "";
        display: block;
        width: 10px;
        height: 10px;
        background-color: #fff;
        border-radius: 50%;
        border: 3px solid #ec720b;
        position: absolute;
        left: 0px;
        z-index: ${zIndexes.STRATOSPHERE};
      }
    }
  `}`;

export const ProgressBarContainer = styled.div`
  display: flex;
  align-items: center;

  p {
    margin-right: 10px;
  }
`;
