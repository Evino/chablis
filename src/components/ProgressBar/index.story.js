import React from 'react';
import { storiesOf } from '@storybook/react';
import SetProgressBar from './SetComponent';

storiesOf('ProgressBar', module).addWithInfo('ProgressBar', () => (
  <SetProgressBar active="3" label="DoÃ§ura" />
));
