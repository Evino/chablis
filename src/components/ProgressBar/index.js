import ProgressBarStyled from './ProgressBarStyled';
import ProgressBarContainer from './ProgressBarContainer';
import Bar from './Bar';

export { Bar, ProgressBarContainer, ProgressBarStyled };
