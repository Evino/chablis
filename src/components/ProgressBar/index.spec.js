import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import ProgressBar from './SetComponent';

const label = 'Teste';
const active = 3;
const element = <ProgressBar label={label} active={active} />;
const component = mount(element);

describe('ProgressBar', () => {
  it('Should render the component', () => {
    const tree = renderer.create(element).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should display passed label', () => {
    expect(component.prop('label')).toBe(label);
  });

  it('Should activate specific active tags', () => {
    const tree = renderer.create(element).toJSON();
    expect(tree).toMatchSnapshot();
    expect(component.prop('active')).toBe(3);
  });
});
