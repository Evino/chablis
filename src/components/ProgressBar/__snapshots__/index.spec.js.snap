// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar Should activate specific active tags 1`] = `
<div
  className="sc-htpNat hLMOoA"
>
  <p
    className="sc-bxivhb hhQZZW"
  >
    Teste
  </p>
  <div
    className="sc-bwzfXH Mxefn"
  >
    <span
      className="sc-bdVaJa eGlyNT"
    />
    <span
      className="sc-bdVaJa eGlyNT"
    />
    <span
      className="sc-bdVaJa eGlyNT"
    />
    <span
      className="sc-bdVaJa eGlyNT"
    />
    <span
      className="sc-bdVaJa eGlyNT"
    />
  </div>
</div>
`;

exports[`ProgressBar Should render the component 1`] = `
<div
  className="sc-htpNat hLMOoA"
>
  <p
    className="sc-bxivhb hhQZZW"
  >
    Teste
  </p>
  <div
    className="sc-bwzfXH Mxefn"
  >
    <span
      className="sc-bdVaJa eGlyNT"
    />
    <span
      className="sc-bdVaJa eGlyNT"
    />
    <span
      className="sc-bdVaJa eGlyNT"
    />
    <span
      className="sc-bdVaJa eGlyNT"
    />
    <span
      className="sc-bdVaJa eGlyNT"
    />
  </div>
</div>
`;
